go-controller/pkg/util/multi_network.go

```diff
@@ -662,19 +662,52 @@ func GetPodNADToNetworkMapping(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*
 	for _, network := range allNetworks {
 		nadName := GetNADName(network.Namespace, network.Name)
 		if nInfo.HasNAD(nadName) {
+			if nInfo.IsPrimaryNetwork() {
+				return false, nil, fmt.Errorf("unexpected primary network %q specified with a NetworkSelectionElement %+v", nInfo.GetNetworkName(), network)
+			}
 			if _, ok := networkSelections[nadName]; ok {
 				return false, nil, fmt.Errorf("unexpected error: more than one of the same NAD %s specified for pod %s",
 					nadName, podDesc)
 			}
 			networkSelections[nadName] = network
 		}
 	}
+
 	if len(networkSelections) == 0 {
 		return false, nil, nil
 	}
+
 	return true, networkSelections, nil
 }
 
+func GetPodNADToNetworkMappingWithActiveNetwork(pod *kapi.Pod, nInfo NetInfo, activeNetwork NetInfo) (bool, map[string]*nettypes.NetworkSelectionElement, error) {
+	_, networkSelections, err := GetPodNADToNetworkMapping(pod, nInfo)
+	if err != nil {
+		return false, nil, err
+	}
+
+	// Add the active network to the NSE map if it is configured
+	if activeNetwork != nil && activeNetwork.GetNetworkName() == nInfo.GetNetworkName() && (nInfo.TopologyType() == types.Layer2Topology || nInfo.TopologyType() == types.Layer3Topology) {
+		activeNetworkNADs := activeNetwork.GetNADs()
+		if len(activeNetworkNADs) < 1 {
+			return false, nil, fmt.Errorf("missing NADs at active network '%s' for namesapce '%s'", activeNetwork.GetNetworkName(), pod.Namespace)
+		}
+		activeNetworkNADKey := strings.Split(activeNetworkNADs[0], "/")
+		if len(networkSelections) == 0 {
+			networkSelections = map[string]*nettypes.NetworkSelectionElement{}
+		}
+		networkSelections[activeNetworkNADs[0]] = &nettypes.NetworkSelectionElement{
```

Comment:
can we export `activeNetworkNADs[0]` to a variable please ? Maybe even ensure it is never an empty list (return an error if it is ...) 

Or are we 100% sure `GetNADs` cannot return an empty slice ?

Reply:
We have a check already 
https://github.com/ovn-org/ovn-kubernetes/pull/4473/files#diff-204cead8f2f1e61a6569211a8c762e70d43a88b06137e1a3cb2eae7ae0a57786R692
```golang
if len(activeNetworkNADs) < 1 {
			return false, nil, fmt.Errorf("missing NADs at active network '%s' for namesapce '%s'", activeNetwork.GetNetworkName(), pod.Namespace)
		}
```

Reply:
Also the unit test we have is already checking that we call NAD's `List` passing the namespace 

```golang
 nadNamespaceLister := v1nadmocks.NetworkAttachmentDefinitionNamespaceLister{}
            nadLister.On("NetworkAttachmentDefinitions", tc.namespace).Return(&nadNamespaceLister)

```

This means that the NADs for same network but at different namespace do not "populate" the function.

Reply:
Good enough then. Tks.